src/utils/evm/test-routes.ts

```diff
@@ -0,0 +1,339 @@
+                  ))
+
+                  break
+                case ResourceType.PERMISSIONLESS_GENERIC:
+                  const messageTransfer = new EVMGenericMessageTransfer()
+
+                  await messageTransfer.init(
+                    sourceProvider as unknown as any,
+                    Environment[environment]
+                  )
+
+                  const EXECUTION_DATA = utils.defaultAbiCoder.encode(
+                    ['uint'],
+                    [Date.now()]
+                  )
+
+                  const genericTransfer = messageTransfer.createGenericMessageTransfer(
+                    await evmWallet.getAddress(),
+                    destinationDomain.chainId,
+                    resource.resourceId,
+                    executionContractAddresses[`EXECUTE_CONTRACT_ADDRESS_${destinationDomain.id}`],
+                    EXECUTE_FUNCTION_SIGNATURE,
+                    EXECUTION_DATA,
+                    MAX_FEE
+                  )
+
+                  const genericFee = await messageTransfer.getFee(genericTransfer)
+                  const genericTransferTx =
+                    await messageTransfer.buildTransferTransaction(
+                      genericTransfer,
+                      genericFee
+                    )
+
+                  const contractValueBefore = await fetchGenericContractValue(
+                    executionContractAddresses[`EXECUTE_CONTRACT_ADDRESS_${destinationDomain.id}`],
+                    destinationProvider,
+                    await evmWallet.getAddress()
+                  )
+
+                  await evmWallet.sendTransaction(
+                    genericTransferTx as providers.TransactionRequest
+                  )
+
+                  functionCalls.push(waitUntilBridgedGenericEvm(
+                    loggingData,
+                    result,
+                    contractValueBefore,
+                    executionContractAddresses[`EXECUTE_CONTRACT_ADDRESS_${destinationDomain.id}`],
+                    destinationProvider,
+                    await evmWallet.getAddress()
+                  ))
+                  break
+                case ResourceType.NON_FUNGIBLE:
+                  print.warning("not implemented for type: " + ResourceType.NON_FUNGIBLE);
+                  break;
+                case ResourceType.PERMISSIONED_GENERIC:
+                  print.warning(
+                    'not implemented for type: ' + ResourceType.NON_FUNGIBLE
+                  )
+                  break
+                default:
+                  print.error(`INVALID RESOURCE TYPE`)
+              }
+            }
+          }
+
+        } catch (err) {
+          print.error(err)
+          result += `\n resource ${loggingData.resourceId} unable to bridged from domain ${loggingData.sourceDomainId}(${loggingData.sourceDomainName}) to domain ${loggingData.destinationDomainId}(${loggingData.destinationDomainName}) - FAILED \n`
+        }
+      }
+
+      // EVM -> SUBSTRATE
```

Comment:
Maybe makes more sense to have testEVMToEVMROutes and testEVMToSubstrateRoutes instead of everything in one